crates/driver/src/infra/api/quote/mod.rs

```diff
@@ -12,8 +12,8 @@ async fn quote(
     let order = order.0.into_domain(state.now()).unwrap();
     let liquidity = state.liquidity().for_quote(&order).await.unwrap();
     let quote = order
-        .quote(state.solver(), &liquidity, state.now())
+        .quote(state.eth(), state.solver(), &liquidity, state.now())
```

Comment:
It feels like `quote` has too many parameters...

I think we could group everything into a:
```
struct Quoter {
  eth: Ethereum,
  solver: Solver,
  liquidity: liquidity::Fetcher,
  now: time::Now,
}
```

Which would play a similar role as the `competition::Competition` type, and keep all quoting-specific state in a single value. WDYT? Just not sure what to call it though...

We could call it `Quote` and rename the existing `Quote` to `Swap`?

Reply:
I'm fine with the number of parameters. The only goal of this struct would be to collect the parameters of the Order::quote function, right? In that case the most natural name to me is QuoteParameters.

Reply:
> The only goal of this struct would be to collect the parameters of the Order::quote function, right?

I would probably invert the call though:
```
let thing = quote::Thing { ... };
let order = quote::Order { ... };
thing.quote(&order).await;
```

Reply:
Naming that `Thing` is tricky. One option could be `QuoteProvider` or something like that but that doesn't tell you much.
Alternatively I thought `Router` might fit, although technically it's not just executing a predetermined trade route but computing it in the first place.

Reply:
I like `Router`.

Reply:
I don't think I like `Router`. Here's a little tidbit from DDD applied to the OO world:

DDD recognizes domain objects which always model abstractions, but also many DDD authors add an additional abstraction layer called the Application Layer where the services reside. Service types are types which don't really have a single representation as a domain object, but rather they represent processes. So these are things like `Quoter`, `PriceEstimator`, and all the other `-er`s that don't belong to the infrastructure layer.

My personal opinion on this in the context of Rust: I don't think that such service types are idiomatic, I think that simply using a function is better (why `quoter.quote()` when you can just say `quote(...)`?) and if you start having too many parameters (roughly the point where clippy starts warning you), try to figure out better domain objects.

Reply:
OTOH `Competition` does model a relatively coherent abstraction: the competition process.

Reply:
But if you insist on having a type to model this process, I think the only good name is `Quoter`.

Reply:
Also https://github.com/cowprotocol/services/issues/1122 can help reduce the parameters by 1 since the `Now` type is a bit ugly to begin with.

Reply:
Left it out for now.